src/account_manager/builder.rs

```diff
@@ -134,18 +134,20 @@ impl AccountManagerBuilder {
         {
             let manager_builder = storage_manager.lock().await.get_account_manager_data().await.ok();
             let (client_options, secret_manager, coin_type) = match manager_builder {
-                Some(data) => {
+                Some(ref data) => {
                     // prioritise provided client_options and secret_manager over stored ones
                     let client_options = match self.client_options {
-                        Some(options) => options,
+                        Some(ref options) => options.clone(),
                         None => data
                             .client_options
+                            .clone()
                             .ok_or(crate::Error::MissingParameter("client_options"))?,
                     };
                     let secret_manager = match self.secret_manager {
-                        Some(secret_manager) => secret_manager,
+                        Some(ref secret_manager) => secret_manager.clone(),
                         None => data
                             .secret_manager
+                            .clone()
```

Comment:
Do we actually need the clones here now since the replace got moved?

Reply:
yes couldn't get it working when removing that

Reply:
@samuel-rufi didn't want to annoy you, pls review https://github.com/samuel-rufi/wallet.rs/pull/123 :trollface:

Reply:
Lgtm